src/components/MatrixBlock/index.js

```diff
@@ -0,0 +1,44 @@
+import { CustomButton } from "../shared/CustomButton";
+import styles from "./MatrixBlock.module.scss";
+import { TableCell } from "../Matrix/TableCell";
+import { TableRow } from "../Matrix/TableRow";
+import { MatrixAverage } from "../Matrix/AvarageRow";
+import { createMatrixRow } from "../../utils/helpers/create-matrix-row";
+
+export const Matrix = ({ data, settings, onSetMatrix }) => {
+  const handleAddRow = () =>
+    onSetMatrix([...data, createMatrixRow(settings.columns)]);
+
+  const getHeadings = (columns) =>
+    Array(columns)
+      .fill(0)
+      .map((_, index) => index + 1);
+
+  const headings = getHeadings(settings.columns);
+
+  return (
+    !!data.length && (
+      <div className="app__matrix-block">
+        <CustomButton data="Add row" onClick={handleAddRow} />
+        <table className={styles.table}>
+          <thead>
+            <tr>
+              <TableCell data="№" />
+              {headings.map((item) => (
+                <TableCell data={item} />
+              ))}
+              <TableCell data="Sum" />
+              <TableCell />
+            </tr>
+          </thead>
+          <tbody>
+            {data.map((item, index) => (
+              <TableRow data={item.row} matrix={data} id={item.id} index={index} styles={styles}   onSetMatrix={onSetMatrix} />
```

Comment:
no need styles and id props. u can use an item.id inside a data row

Reply:
Why don`t we need styles here, if we use styles inside the component?